### ISSUE TITLE
Can't accept traffic in INVALID state ?

### ISSUE THREAD

**Issue by bluejam0303** [firehol/firehol#208]
Hi all,

In some scenario, our server close connection actively but I find the socket staying in FIN_WAIT2.
After investigating, I find it is due to the FIN packet from client dropped by iptables rule.
I have already accepted traffic for our service port in Firehol rule but I find it only accept traffic in state "NEW,ESTABLISHED".
I tried to add something like following to accept packet in the state "NEW,ESTABLISHED,INVALID" for our port but it is warning and overwritten by  "NEW,ESTABLISHED"

# setting in firehol config
server custom test  tcp/8080  any  accept  state NEW,ESTABLISHED,INVALID

# result
...
WHY    : Overwritting param: state 'NEW,ESTABLISHED,INVALID' becomes 'NEW,ESTABLISHED'
...

Could I know if we could accept traffic in INVALID state and how to config it ?

Thanks,
Jeffery

**Comment by philwhineray** [comment]
I don't recommend allowing invalid packets if possible.

I believe that the netfilter connection tracker moves connections out of
its tables early when it is under pressure. I.e. it would not normally mark
these packets as invalid but it is trying to reclaim space.

Have you checked dmesg/syslog to see if there is any information in there?

If this is the problem you would be better tuning the conntrack. Try
https://wiki.khnet.info/index.php/Conntrack_tuning

**Comment by bluejam0303** [comment]
Hi philwhineray

I am able to reproduce it in our dev host with less traffic and I don't see any abnormal log from dmesg and syslog. So I think it may be not related to you mentioned.

From my testing tcpdump result, it is like following,

server > client [F]   
client > server [A]    
client > server [F]  (FIN packet from client to our server continuously but no responses from our server to client. In the normal situation, our server should ACK the FIN packet and close the connection normally)

I tried to add accepting traffic in INVALID state by iptables manually and the issue is also gone.
I know allowing invalid packets is not a good idea but it looks it indeed caused by the rule.
If I just want to allow packet in INVALID state, do I only add custom iptable rules in Firehol config file instead of adding  state "NEW,ESTABLISHED,INVALID" after "accept" command ?

Thanks,
Jeffery

**Comment by philwhineray** [comment]
Yes, as you've discovered, firehol will overwrite the states in a standard
composition but there is nothing stopping you from including a fully custom
iptables statement to insert a specific rule.
